### PR TITLE
TST: fix assert_categorical_equal message

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] stage 1 of #13076
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] no whatsnew as it is internal test method

```
a = pd.Categorical([1, 2, 3, 4])
b = pd.Categorical([1, 2, 3, 5])
tm.assert_categorical_equal(a, b)
# Categorical.categories are different
# ...

a = pd.Categorical([1, 2, 4, 3], categories=[1, 2, 3, 4])
b = pd.Categorical([1, 2, 3, 4], categories=[1, 2, 3, 4])
tm.assert_categorical_equal(a, b)
# Categorical.codes values are different (50.0 %)
# ...
```

This must be after #13088.

### CODE REVIEW

File: pandas/util/testing.py [hunk 1]
```diff
@@ -887,18 +887,16 @@ def assertNotIsInstance(obj, cls, msg=''):
...
-    assertIsInstance(res, pd.Categorical, '[Categorical] ')
-    assertIsInstance(exp, pd.Categorical, '[Categorical] ')
+def assert_categorical_equal(left, right, obj='Categorical'):
+    assertIsInstance(left, pd.Categorical, '[Categorical] ')
+    assertIsInstance(right, pd.Categorical, '[Categorical] ')
 
-    assert_index_equal(res.categories, exp.categories)
+    assert_index_equal(left.categories, right.categories,
+                       obj='{0}.categories'.format(obj))
+    assert_numpy_array_equal(left.codes, right.codes,
```
Comment (jreback) [inline pandas/util/testing.py]:
```
In [2]: tm.assert_numpy_array_equal(np.array([1,2],dtype='int64'),np.array([1,2],dtype='int8'))
Out[2]: True
```
check the dtypes are equal as well (in fact we might want to do that in ``assert_numpy_array_equal``)

File: pandas/util/testing.py [hunk 2]
```diff
@@ -887,18 +887,16 @@ def assertNotIsInstance(obj, cls, msg=''):
...
+                             obj='{0}.codes'.format(obj))
 
-    if not array_equivalent(res.codes, exp.codes):
-        raise AssertionError(
-            'codes not equivalent: {0} vs {1}.'.format(res.codes, exp.codes))
-
-    if res.ordered != exp.ordered:
-        raise AssertionError("ordered not the same")
+    assert_attr_equal('ordered', left, right, obj=obj)
```
Comment (jreback) [inline pandas/util/testing.py]: hmm, why didn't this work before?

Comment (sinhrks) [inline pandas/util/testing.py]: The change is for better error reporting purpose.

As described in #13076, the check itself is OK but it is not called from Series/Index check, no?

Comment (jreback) [inline pandas/util/testing.py]: ahh I c, ok then.

File: pandas/util/testing.py (hunk 1 above)
Comment (sinhrks) [inline pandas/util/testing.py]: Will split this to other PR as there are lots of failures (unrelated to categorical).

File: pandas/core/categorical.py [hunk 3]
```diff
@@ -985,7 +985,7 @@ def __setstate__(self, state):
 
         # Provide compatibility with pre-0.15.0 Categoricals.
         if '_codes' not in state and 'labels' in state:
-            state['_codes'] = state.pop('labels')
+            state['_codes'] = state.pop('labels').astype(np.int8)
```
Comment (sinhrks) [inline pandas/core/categorical.py]: Minor fix to make `dtype` compat. Needs whatsnew?

**Comment by jreback** [comment]
ty sir!

**Pull request closed without merging**